### PR TITLE
chore: fcm app fixes for ci

### PR DESCRIPTION
**Pull request by mrehan27**
part of [MBL-421](https://linear.app/customerio/issue/MBL-421/react-native-sample-apps-to-firebase-distribution)

### Changes

- Removed `package-lock.json` from `.gitignore` so `npm ci` can be used in Github actions
- Added `ios/Env.swift.example` to FCM sample
- Added `ci:install` action so both sample apps can run scripts independently and it can be used by Github action to build app
- Updated project and lock files so FCM can be built using Github actions

### CODE REVIEW

File: Apps/FCM/package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "start": "react-native start",
     "dev:update": "../../scripts/dev:update.sh",
     "postinstall": "pod update --project-directory=ios",
-    "preinstall": "npm run dev:update"
+    "preinstall": "npm run dev:update",
+    "ci:install": "npm run preinstall && npm ci"
```
Comment (mrehan27) [inline Apps/FCM/package.json:10]: This will be used by CI to install dependencies

**Review by ami-aman — APPROVED** [review]
Looks good 👍

File: Apps/FCM/ios/FCMSampleApp/PrivacyInfo.xcprivacy
Comment (Shahroz16) [inline Apps/FCM/ios/FCMSampleApp/PrivacyInfo.xcprivacy:1]: interesting i thought this was only needed when you want to push to app store

**Review by Shahroz16 — APPROVED** [review]

File: Apps/FCM/ios/FCMSampleApp/PrivacyInfo.xcprivacy
Comment (mrehan27) [reply Apps/FCM/ios/FCMSampleApp/PrivacyInfo.xcprivacy:1]: This is automatically generated when you install pods and run the app.

**Pull request merged**